### PR TITLE
add ability to set the Content-Type for the uploaded chart.

### PR DESCRIPTION
**Pull request by bashims**
#59 Add ability to optionally specify the Content-Type for the uploaded chart.

### CODE REVIEW

**Comment by bashims** [comment]
I noticed that the shell implementation of the integration tests are "legacy", so I ported the same tests to test_push.go.

File: cmd/helms3/main.go [hunk 1]
```diff
@@ -87,6 +87,10 @@ func main() {
 		Bool()
 	pushIgnoreIfExists := pushCmd.Flag("ignore-if-exists", "If the chart already exists, exit normally and do not trigger an error.").
 		Bool()
+	pushContentType := pushCmd.Flag("content-type", "").
+		Default("").
```
Comment (kylehodgetts) [inline cmd/helms3/main.go]: A sensible default should probably be set here

**Comment by bashims** [comment]
Hi @kylehodgetts  I added the default content-type. Can you take another crack at the review?

Thanks,

-ben

**Comment by kylehodgetts** [comment]
Hey. It looks good to me but I'm just a user :) 

The maintainer of the repo @hypnoglow will need to okay this.

**Comment by bashims** [comment]
Hi @hypnoglow ,

Would you be able to take a look at his PR, we'd like to start using in your deployment pipeline.

Thanks,

-ben

File: cmd/helms3/main.go [hunk 2]
```diff
@@ -24,6 +24,9 @@ const (
 	defaultTimeout       = time.Minute * 5
 	defaultTimeoutString = "5m"
 
+	// duplicated in e2e package for testing.
+	defaultChartsContentType = "application/x-gzip"
```
Comment (hypnoglow) [inline cmd/helms3/main.go]: Should we use `application/gzip` instead, as defined per [RFC 6713](https://tools.ietf.org/html/rfc6713)? Also: https://www.iana.org/assignments/media-types/media-types.xhtml

**Comment by hypnoglow** [comment]
Awesome, thanks @bashims!

Please check the comment about default content type. Everything else looks good.

File: cmd/helms3/main.go (hunk 2 above)
Comment (bashims) [reply cmd/helms3/main.go]: Yes, x-gzip is deprecated.

**Comment by hypnoglow** [comment]
Thanks again @bashims

**Comment by bashims** [comment]
No problem. Thanks for getting this released so quickly!

**Pull request merged**